### PR TITLE
feat/refactor/docs(observability): epoch hook, txfees epoch, existing metric clean up

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

- Adds counters for txfees epoch flow
- Adds counters for epoch hook panics
- Cleans up and simplifies the existing counters
- Removes superfluous logs
- Adds detailed docs

### CODE REVIEW

File: x/concentrated-liquidity/incentives.go [hunk 1]
```diff
@@ -260,7 +260,7 @@ func calcAccruedIncentivesForAccum(ctx sdk.Context, accumUptime time.Duration, l
 		// Total amount emitted = time elapsed * emission
 		totalEmittedAmount, err := computeTotalIncentivesToEmit(timeElapsed, incentiveRecordBody.EmissionRate)
 		if err != nil {
-			ctx.Logger().Info(types.IncentiveOverflowTelemetryName, "pool_id", poolID, "incentive_id", incentiveRecord.IncentiveId, "time_elapsed", timeElapsed, "emission_rate", incentiveRecordBody.EmissionRate, "error", err.Error())
+			emitIncentiveOverflowTelemetry(poolID, incentiveRecord.IncentiveId, timeElapsed, incentiveRecordBody.EmissionRate, err)
```
Comment (p0mvn) [inline x/concentrated-liquidity/incentives.go:263]: @alessandrolomanto please note that I removed logs, leaving only Prometheus counters.

Counters with labels should be sufficient for us to set up alerts with enough context. Having these logs could  cause node log pollution

File: x/concentrated-liquidity/incentives.go [hunk 2]
```diff
@@ -1035,22 +1035,6 @@ func (k Keeper) getLargestSupportedUptimeDuration() time.Duration {
...
-		ctx.Logger().Error(truncatedPlaceholder, "pool_id", poolID, "total_liq", liquidityInAccum, "per_unit_liq", rewardsPerUnitOfLiquidity, "total_amt", rewardsTotal, extraKeyVals)
-
-		// We emit events for these pools specifically as they are at the border of truncation in terms of liquidity
-		// TODO: remove these after scaling factor approach is implemented
-	} else if poolID == (1423) || poolID == (1213) {
-		telemetry.IncrCounter(1, emittedPlaceholder)
-		ctx.Logger().Info(emittedPlaceholder, "pool_id", poolID, "total_liq", liquidityInAccum, "per_unit_liq", rewardsPerUnitOfLiquidity, "total_amt", rewardsTotal, extraKeyVals)
-	}
-}
-
```
Comment (p0mvn) [inline x/concentrated-liquidity/incentives.go:1053]: Note: moved into a separate file

File: x/epochs/types/hooks.go [hunk 3]
```diff
@@ -13,29 +16,41 @@ type EpochHooks interface {
...
+
 var _ EpochHooks = MultiEpochHooks{}
 
 // combine multiple gamm hooks, all hook functions are run in array sequence.
 type MultiEpochHooks []EpochHooks
 
+// GetModuleName implements EpochHooks.
+func (MultiEpochHooks) GetModuleName() string {
+	return ModuleName
+}
```
Comment (p0mvn) [inline x/epochs/types/hooks.go:36]: Note: the core of the epoch modification is to expose this method for additional context with telemetry. It is state-compatible

File: x/epochs/types/telemetry.go [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+var (
+	// epoch_hook_failed
+	//
+	// counter that is increased if epoch hook fails
+	//
+	// Has the following labels:
+	// * module_name - the name of the module that errored or panicked
+	// * err - the error or panic returned
+	// * is_before_hook - true if this is a before epoch hook. False otherwise.
+	EpochHookFailedMetricName = formatEpochMetricName("hook_failed")
```
Comment (p0mvn) [inline x/epochs/types/telemetry.go:14]: @alessandrolomanto could you please review all `telemetry.go` update files in this PR and configure alerts for each?

If the counter increases, it should send an alert. Please also format labels in the alert

**Review by ValarDragon — APPROVED** [review]
This LGTM, though I am worried that in the SDK we found that telemetry was actually a real cost to sync speed, so we had to do things to make it only get emitted/computed if telemetry was enabled. Maybe thats inherited by us right now as well? Unsure

**Comment by p0mvn** [comment]
> This LGTM, though I am worried that in the SDK we found that telemetry was actually a real cost to sync speed, so we had to do things to make it only get emitted/computed if telemetry was enabled. Maybe thats inherited by us right now as well? Unsure

I think the changes in this PR are unlikely to be a problem for sync speed for the following reasons:
1. Telemetry needs to be explicitly enabled
2. We emit it mostly in error cases that we do not expect to occur for alerting purposes.

I'm going to merge this to keep our alert efforts going but please let me know if any follow-up concerns

**Pull request merged**